src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+        func_800B874C(&this->actor, globalCtx, 100.0f, 80.0f);
+    }
+}
+
+void EnKakasi_SetupDialogue(EnKakasi* this) {
+    // bug? starts one animation then changes it a few lines down?
+    if (this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+    }
+
+    this->unkMsgState1AC = 5;
+    this->unkState196 = 1;
+    EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SPIN_REACH_OFFER);
+    this->actionFunc = EnKakasi_RegularDialogue;
+}
+
+void EnKakasi_RegularDialogue(EnKakasi* this, GlobalContext* globalCtx) {
+    u32 saveContextDay = gSaveContext.day;
+    f32 currentAnimeFrame = this->skelanime.animCurrentFrame;
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 2000, 0);
+    // "Yo! Hey, baby!  I'm a stylin' scarecrow wandering in search of pleasant music.
+    // Time will pass in the blink of an eye if you dance with me."
+    if (this->actor.textId != 0x1644 && this->animeFrameCount <= currentAnimeFrame &&
+        this->animeIndex == ENKAKASI_ANIME_SLOWROLL) {
+
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SPIN_REACH_OFFER);
+        this->unkCounter1A4 = 0;
+    }
+
+    // "Oh, sorry. That's too bad. In that case, come back anytime if you do ...
+    // "How's that? Pretty interesting, isn't it?  But I'm sure I'd like any song
+    if (this->actor.textId == 0x1651 || this->actor.textId == 0x1659) {
+        if (this->animeFrameCount <= currentAnimeFrame && this->animeIndex != ENKAKASI_ANIME_SPIN_REACH_OFFER) {
+            if (++this->unkCounter1A4 >= 2) {
+                this->unkCounter1A4 = 0;
+                EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SPIN_REACH_OFFER);
+            }
+        }
+    }
+
+    if (this->unkState1A8 == 2 && this->unkState196 == 2) {
+        func_800B7298(globalCtx, &this->actor, 0x49);
+        this->unkState1A8 = 0;
+    }
+
+    if (this->unkMsgState1AC == func_80152498(&globalCtx->msgCtx) && func_80147624(globalCtx) != 0) {
+        func_801477B4(globalCtx);
+        if (this->unkMsgState1AC == 5) {
+            // "No, no, daddy-o!  That song doesn't sing to my heart, baby!"
+            if (this->unkState196 == 2 && this->actor.textId == 0x1647) {
+                func_800B7298(globalCtx, &this->actor, 6);
+            }
+
+            // "Oh, yeah!  How was it? It went by in an instant, right?  I'm still full of energy!"
+            if (this->actor.textId == 0x1653) {
+                u32 saveContextDay2 = gSaveContext.day;
+
+                if (this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+                    EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+                }
+
+                if (saveContextDay2 == 3 && gSaveContext.isNight != 0) {
+                    // "But outside it seems to have gotten kind of...dangerous.
+                    this->actor.textId = 0x164F;
+                } else if (gSaveContext.isNight != 0) {
```

Comment:
```suggestion
                } else if (gSaveContext.isNight) {
```
I'd guess `isNight` is a bool?